src/main/java/com/google/research/bleth/simulator/StatisticsState.java

```diff
@@ -41,10 +46,30 @@
      * @return a new instance of StatisticsState
      */
     public static StatisticsState create(String simulationId, Map<String, Double> distanceStats,
-                                         Table<String, String, Double> beaconsObservedStats) {
+                                         Table<String, String, Double> beaconsObservedStats,
+                                         LinkedListMultimap<Integer, ObservedInterval> beaconsObservedIntervals) {
         checkNotNull(distanceStats);
         checkNotNull(beaconsObservedStats);
-        return new StatisticsState(simulationId, ImmutableMap.copyOf(distanceStats), beaconsObservedStats);
+        checkNotNull(beaconsObservedIntervals);
+        return new StatisticsState(simulationId, ImmutableMap.copyOf(distanceStats),
+                   beaconsObservedStats, beaconsObservedIntervals);
+    }
+
+    /** Create and write datastore entities storing beacons' observed intervals. */
+    public void writeIntervalStats() {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+        for (Integer beaconId : beaconsObservedIntervals.keySet()) {
+            for (ObservedInterval interval : beaconsObservedIntervals.get(beaconId)) {
+                Entity entity = new Entity(Schema.StatisticsState.entityKindBeaconsObservedIntervals);
+                entity.setProperty(Schema.StatisticsState.simulationId, simulationId);
+                entity.setProperty(Schema.StatisticsState.beaconId, beaconId);
+                entity.setProperty(Schema.StatisticsState.intervalStart, interval.start());
+                entity.setProperty(Schema.StatisticsState.intervalEnd, interval.end());
+                entity.setProperty(Schema.StatisticsState.intervalObserved, interval.observed() ? 1 : 0);
```

Comment:
Why are you using 1/0, and not something more descriptive (e.g. a string "OBSERVED" / "NOT_OBSERVED")?